lib/pos_blockchain.go

```diff
@@ -1898,19 +1938,16 @@ func (bc *Blockchain) getUtxoViewAtBlockHash(blockHash BlockHash) (*UtxoView, er
 				"getUtxoViewAtBlockHash: extends from a committed block that isn't the committed tip")
 		}
 	}
-	if viewAtHash, exists := bc.blockViewCache.Lookup(blockHash); exists {
-		copiedView, err := viewAtHash.(*UtxoView).CopyUtxoView()
-		if err != nil {
-			return nil, errors.Wrapf(err, "getUtxoViewAtBlockHash: Problem copying UtxoView from cache")
+	if viewAndUtxoOpsAtHashItem, exists := bc.blockViewCache.Lookup(blockHash); exists {
+		viewAndUtxoOpsAtHash, ok := viewAndUtxoOpsAtHashItem.(*BlockViewAndUtxoOps)
+		if !ok {
+			return nil, errors.New("getUtxoViewAtBlockHash: Problem casting to BlockViewAndUtxoOps")
 		}
```

Comment:
Wrap the blockViewCache with a helper function that internally performs the cast to the `*BlockViewAndUtxoOps` type. It's fine to call the helper function something like `getCachedBlockViewAndUtxoOps`
```suggestion
	if viewAndUtxoOpsAtHash, exists := bc.getCachedBlockViewAndUtxoOps(blockHash); exists {
		...
```